node/node.go

```diff
@@ -381,8 +384,9 @@ func (node *AlgorandFullNode) startMonitoringRoutines() {
 	node.monitoringRoutinesWaitGroup.Add(2)
 	go node.txPoolGaugeThread(node.ctx.Done())
 	// Delete old participation keys
-	go node.oldKeyDeletionThread(node.ctx.Done())
-
+	pprof.Do(context.Background(), pprof.Labels("worker", "oldKeyDeletionThread"), func(_ context.Context) {
```

Comment:
Why doesn't the txPoolGaugeThread get labeled?

Reply:
It doesn't do much work ... I guess I should wrap it up for completeness though, it calls into the TransactionPool to ask for its size, so I might miss something if that is expensive.

Reply:
Is it possible for a thread/goroutine to label itself? It seems slightly better for the go routine itself to have a "label me" line as its first line of code, than to change call sites.

Reply:
I was thinking that too. the official pprof docs say to always use the `Do` wrapper but I think calling the [pprof.SetGoroutineLabels](https://pkg.go.dev/runtime/pprof#SetGoroutineLabels) function directly inside the goroutine itself is a little nicer-looking for us.